include/unifex/unstoppable.hpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2019-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/tag_invoke.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/type_traits.hpp>
+#include <unifex/type_list.hpp>
+#include <unifex/manual_lifetime.hpp>
+#include <unifex/manual_lifetime_union.hpp>
+#include <unifex/async_trace.hpp>
+
+#include <utility>
+#include <cassert>
+#include <exception>
+
+namespace unifex {
+
+namespace detail {
+
+template<typename Source, typename Receiver>
+class unstoppable_operation;
+
+template<typename Source, typename Receiver>
+class unstoppable_source_receiver;
+
+template<typename Source, typename Receiver>
+class unstoppable_source_receiver {
+  using operation = unstoppable_operation<Source, Receiver>;
+public:
+  explicit unstoppable_source_receiver(operation* op) noexcept
+  : op_(op) {}
+
+  unstoppable_source_receiver(unstoppable_source_receiver&& other) noexcept
+  : op_(std::exchange(other.op_, {}))
+  {}
+ 
+  void set_value() {
```

Comment:
This should be conditionally noexcept if the inner receiver has a noexcept set_value call.